### ISSUE TITLE
Cannot connect to node

### ISSUE THREAD

**Issue by asocjal** [ElementsProject/lightning#1210]
### Issue and Steps to Reproduce
When created channel between both c-lightning version v0.5.2-2016-11-21-2269-g249464c channels worked fine. After some time I realized that nodes are nort connected (listpeers). When trying to connect from cli, I have error:

```
{ "peers" : 
        [ 
                { "id" : "034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4", "connected" : false, "channels" : 
                        [ 
                                { "state" : "ONCHAIN", "short_channel_id" : "513012:1879:0", "funding_txid" : "de62f6cae46f0545e46f4436ba645d697b17961d373b6d9743ecbbd50957a216", "msatoshi_to_us" : 2225000, "msatoshi_total" : 100000000, "dust_limit_satoshis" : 546, "max_htlc_value_in_flight_msat" : 18446744073709551615, "channel_reserve_satoshis" : 0, "htlc_minimum_msat" : 0, "to_self_delay" : 144, "max_accepted_htlcs" : 483, "status" : 
                                        [ "Loaded from database" ] } ] } ] }
```

```
root@cln-02:~/cln/lightning/cli# ./lightning-cli connect 034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4
{ "code" : -1, "message" : "Could not connect to 034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4 after 45 seconds and 10 attempts" }
```

### `getinfo` output
node1:
`{ "id" : "03f8245976b7cbcf8872e9c328abca86f0f42e503b0532430608f11ad73c6bd4a0", "port" : 9735, "address" : 
        [ 
                { "type" : "ipv4", "address" : "80.211.252.104", "port" : 9735 } ], "version" : "v0.5.2-2016-11-21-2269-g249464c", "blockheight" : 513370, "network" : "$itcoin" }`

node2
`{ "id" : "034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4", "port" : 9735, "address" : 
        [ 
                { "type" : "ipv4", "address" : "80.211.252.169", "port" : 9735 } ], "version" : "v0.5.2-2016-11-21-2269-g249464c", "blockheight" : 513352, "network" : "b
itcoin" }
`

Please note, that getinfo node1 has "network" : "$itcoin" instead of "bitcoin". Logs are attached:
[node1.log.txt](https://github.com/ElementsProject/lightning/files/1807881/node1.log.txt)
[node2.log.txt](https://github.com/ElementsProject/lightning/files/1807882/node2.log.txt)

**Comment by cdecker** [comment]
Well that's simply node2 not accepting incoming connections on port 9735, it could either be down, or behind NAT:

```
telnet 80.211.252.169 9735
Trying 80.211.252.169...
telnet: Unable to connect to remote host: Connection refused
```

**Comment by asocjal** [comment]
Because it was switched off. Please try again now.

**Comment by cdecker** [comment]
Why does one node declare its own network as `$itcoin`, I've never seen such a behavior before. It seems however that it still self-identified as `bitcoin` since otherwise the channel could never have been established.

The `Could not connect to 034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4 after 45 seconds and 10 attempts` message stems from it not being able to open a TCP connection, and the disconnection is normal for channels that are being settled on-chain. Indeed the funding transaction was spent by [e83a65d00e3d0ef0e2bb96cefb9d748cf30a7c5ae4d20a58e14fe356bf2bc4ba](https://www.smartbit.com.au/tx/e83a65d00e3d0ef0e2bb96cefb9d748cf30a7c5ae4d20a58e14fe356bf2bc4ba). Since it has a P2WSH output this is a unilateral close, and since the amount matches the owned amount by node2 this close was initiated by node2. Sadly the log seems to start after the channel closure was initiated, so I can't say what caused it.

**Comment by asocjal** [comment]
This is not the first time I got this issue, so maybe I am able to reproduce it with debug logs switched on. Will try. Maybe this is important but exact behavior was that:

1. Call listPeers, I realize that node is connected==false
2. Call "connect" seems to suceeded
3. Call listPeers again, and still connected==false
4. Call disconnect
5. Call listPeers again, and still connected==false
6. Call connect and got error described before
7. Call listPeers again, and still connected==false

Is this possible that funding transaction was closed by c-lightning automatically (because of some connection problems)? I din't call close for this channel.

**Comment by asocjal** [comment]
It happens again. My both nodes disconnected and I cannot connect them again. Full logs attached.

[c-lightning1.log](https://github.com/ElementsProject/lightning/files/1826146/c-lightning1.log)
[c-lightning2.log](https://github.com/ElementsProject/lightning/files/1826147/c-lightning2.log)

**Comment by cdecker** [comment]
The only relevant entry in the logs I could find was this

```
No address known for 034fd0be2b82f609f30593d2c5b7f27eb33c5e84f660446e4bf1501c838ce052d4, giving up
```

Didn't you provide an IP?

**Comment by asocjal** [comment]
Yes, I know that. Maybe it would be easier if you see details of my tests? I marked first node blue and second node green:

https://docs.google.com/document/d/1x9gtsL7uWpuLsTPj9wXLKkCHbwljycRKKnrvQCCermo/edit?usp=sharing

**Comment by cdecker** [comment]
I see, it appears that you ran into #1194: the funding transaction did not meet the `min_relay_fee`, and eventually the nodes failed the channel funding being unable to broadcast the funding transaction.

The error report 
```
496e7465726e616c206572726f723a2046756e64696e672062726f6164636173742065786974656420776974682032363a206572726f7220636f64653a202d32360a6572726f72206d6573736167653a0a36363a206d696e2072656c617920666565206e6f74206d65740a
```
is hex-encode for:
```
Internal error: Funding broadcast exited with 26: error code: -26
error message:
66: min relay fee not met
```

**Comment by cdecker** [comment]
Duplicate of #1194

**Comment by cdecker** [comment]
This should have been fixed by #1251, can you confirm that newly funded channels no longer have this issue?
